### PR TITLE
[CM H3] added link checker jni plugin

### PR DESCRIPTION
**Pull request by aaronabebe**
I added a plugin called by `check/link` that checks a given link by either calling it or, if no internet connection is available, will try to validate the link by using a regex. 
Also added unit tests. 

Please ignore the other changes except the ones in ` src/bindings/jna/libelektra`. The other changes stem from a merging/rebasing problem in #3801 I wasn't able to solve yet.

I will also update this PR once I fix the other problems in my forks `master` branch. 
Also there are currently build problems when writing jni plugins, so I will have to update the PR anyway.

## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/news/_preparation_next_release.md
Comment (markus2330) [inline doc/news/_preparation_next_release.md:186]: Something went wrong with the rebase.

File: src/bindings/jna/libelektra/dependencies.gradle
Comment (markus2330) [inline src/bindings/jna/libelektra/dependencies.gradle:8]: @tucek any thoughts about which test framework we want to use?

File: src/bindings/jna/libelektra/src/main/java/org/libelektra/plugin/LinkPlugin.java
Comment (markus2330) [inline src/bindings/jna/libelektra/src/main/java/org/libelektra/plugin/LinkPlugin.java:68]: The specification should decide what is required (only regex check or additionally reachable check).

**Review by markus2330 — COMMENTED** [review]
Please rebase and fix build server issues.

**Comment by kodebach** [comment]
> Please ignore the other changes except the ones in `src/bindings/jna/libelektra`. The other changes stem from a merging/rebasing problem in #3801 I wasn't able to solve yet.
> I will also update this PR once I fix the other problems in my forks `master` branch.

I think the problem, is that you used your fork's `master` branch for #3801. We recommend keeping the `master` branch in sync with the main repo, because otherwise you easily get problems like this. See also ["Working with forks" in our git docs](https://github.com/ElektraInitiative/libelektra/blob/master/doc/GIT.md#working-with-forks)

Please create a new branch for #3801 in your fork and reset the `master` to match the `master` of the main libelektra repo. To fix this PR, you will probably have to reset your `CM-H3]` branch to `master` and then cherry-pick the commit that should actually be on the branch from `origin/CM-H3]`.

> Also there are currently build problems when writing jni plugins, so I will have to update the PR anyway.

The problems should now be fixed. Updating your forks `master` to match the main repo's master and then rebasing this PR should fix the problems. Note: You may have to set `ReferenceCleaner.ENABLE_AUTO_NATIVE_REF_CLEANUP = false` to avoid segfaults.

If you have any troubles with the rebase, feel free to tag me in a comment and I will have a look.

File: src/bindings/jna/libelektra/dependencies.gradle
Comment (tucek) [reply src/bindings/jna/libelektra/dependencies.gradle:8]: mockito is usually a good choice in my opinion

**Review by kodebach — CHANGES_REQUESTED** [review]
There are a few minor issue with the plugin. But there is one big problem: The plugin currently only checks a single Key.

PS. If you still have trouble with the segfaults or other problems with the Java API, please tell me, I will try to help you best I can.

**Comment by tucek** [comment]
I work this issue once PR #3889 is merged

**Comment by kodebach** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
@kodebach did you test this with #4239?

**Comment by kodebach** [comment]
No this still uses the old API. I will update it when #4239 is finished

**Comment by kodebach** [comment]
@markus2330 This PR is now updated to work with the new `process` plugin.

**Pull request merged**
